packages/ui/src/models/visualization/flows/support/camel-component-schema.service.ts

```diff
@@ -47,18 +47,23 @@ export class CamelComponentSchemaService {
   }
 
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  static getLabel(camelElementLookup: ICamelElementLookupResult, definition: any): string {
+  static getNodeLabel(camelElementLookup: ICamelElementLookupResult, definition?: any): string {
+    if (typeof definition?.description === 'string' && definition.description !== '') {
+      return definition.description;
+    }
+
     if (camelElementLookup.componentName !== undefined) {
       return camelElementLookup.componentName;
     }
 
+    const uriString = getUriString(definition);
```

Comment:
Streamlined how to extract the `uri` string, considering whether the definition is a string or if it's an object with a `uri` property in it. It also handles empty string cases.